dali/operators/image/resize/tensor_resize_attr.cc

```diff
@@ -0,0 +1,314 @@
+    };
+  } else {
+    DALI_FAIL(make_string(
+        "``rounding`` value ", rounding,
+        " is not supported. Supported values are \"round\", \"truncate\", or \"ceil\"."));
+  }
+}
+
+void TensorResizeAttr::TrimSpatialDims(const TensorListShape<> &input_shape) {
+  int nsamples = input_shape.num_samples();
+  auto unchanged_dim = [&](int d) {
+    for (int i = 0; i < nsamples; i++) {
+      int64_t extent = input_shape.tensor_shape_span(i)[d];
+      if (static_cast<int64_t>(params_[i].dst_size[d]) != extent ||
+          static_cast<int64_t>(params_[i].src_lo[d]) != 0 ||
+          static_cast<int64_t>(params_[i].src_hi[d]) != extent) {
+        return false;
+      }
+    }
+    return true;
+  };
+
+  int can_trim_n = std::max(0, spatial_ndim_ - 2);  // at least 2 spatial dims should remain
+  int new_first_spatial_dim = first_spatial_dim_;
+  int new_end_spatial_dim = first_spatial_dim_ + spatial_ndim_;
+
+  for (int d = first_spatial_dim_; d < first_spatial_dim_ + spatial_ndim_; d++) {
+    if (can_trim_n == 0 || !unchanged_dim(d))
+      break;
+    new_first_spatial_dim++;
+    can_trim_n--;
+  }
+  for (int d = new_end_spatial_dim - 1; d > new_first_spatial_dim; d--) {
+    if (can_trim_n == 0 || !unchanged_dim(d))
+      break;
+    new_end_spatial_dim++;
+    can_trim_n--;
+  }
+
+  int new_spatial_ndim = new_end_spatial_dim - new_first_spatial_dim;
+  if (first_spatial_dim_ == new_first_spatial_dim && spatial_ndim_ == new_spatial_ndim)
+    return;
+
+  for (int s = 0; s < nsamples; s++) {
+    auto &p = params_[s];
+    for (int d = 0; d < new_spatial_ndim ; d++) {
+      int orig_d =  new_first_spatial_dim - first_spatial_dim_ + d;
+      p.dst_size[d] = p.dst_size[orig_d];
+      p.src_hi[d] = p.src_hi[orig_d];
+      p.src_lo[d] = p.src_lo[orig_d];
+    }
+    p.dst_size.resize(new_spatial_ndim);
+    p.src_hi.resize(new_spatial_ndim);
+    p.src_lo.resize(new_spatial_ndim);
+  }
+  first_spatial_dim_ = new_first_spatial_dim;
+  spatial_ndim_ = new_spatial_ndim;
+}
+
+void TensorResizeAttr::PrepareResizeParams(const OpSpec &spec, const ArgumentWorkspace &ws,
+                                           const TensorListShape<> &input_shape,
+                                           const TensorLayout &layout) {
+  input_shape_ = input_shape;
+  (void) input_shape;  // we are potentially modifying our copy for internal use
+  ndim_ = input_shape_.sample_dim();
+  if (ndim_ < 1)
+    throw std::runtime_error("Input must be at least 1D");
+  axes_helper_.PrepareAxes(layout, ndim_);
+  span<int> axes = axes_helper_.Axes();
+  first_spatial_dim_ = 0;
+  // We need spatial_ndim >= 2
+  add_leading_spatial_ndim_ = std::max(0, 2 - ndim_);
+  if (add_leading_spatial_ndim_ > 0) {
+    expand_dims(input_shape_, input_shape, first_spatial_dim_, ndim_ + add_leading_spatial_ndim_);
+    ndim_ = input_shape_.sample_dim();
+    for (int &a : axes)  // skip dummy dims
+      a += add_leading_spatial_ndim_;
+  }
```

Comment:
I think this logic should go to the `TensorResize` class - it's a limitation of a particular implementation, not a generic thing.

Reply:
Done